src/cfgparse.c

```diff
@@ -205,6 +205,13 @@ int str2listener(char *str, struct proxy *curproxy, struct bind_conf *bind_conf,
 			goto fail;
 		}
 	} /* end while(next) */
+#ifdef __linux__
+#ifndef IPPROTO_MPTCP
+#define IPPROTO_MPTCP
+#endif
+	if (global.tune.options & GTUNE_MPTCP && proto->sock_prot == IPPROTO_TCP)
+		proto->sock_prot = IPPROTO_MPTCP;
+#endif
```

Comment:
I'm still not sure how it can work.

Can you move this to the line 202, just before `create_listeners()`? (with an empty line before and after this new block)

Reply:
OK, sorry, I now took some time to dig in the code: `create_listeners()` doesn't create the listerner socket, it only adds it to list (see `default_add_listener()`). The socket is created later, in `bind()`. Here, only the last `proto` is modified: so if by chance, the last socket is a TCP one, it will work. And it works because `proto` will be one item of `__protocol_by_family` array.

In short, this should be moved into the loop.